src/dmd/mtype.d

```diff
@@ -7353,6 +7353,29 @@ bool isCopyable(Type t)
     return true;
 }
 
+/*******************************************
+ * Decide if ref-return-scope should be `ref` `return scope`.
+ * Params:
+ *    returnByRef = if function returns by ref
+ *    stc = storage class to possibly adjust by addition of STC.returnScope
+ */
+void inferReturnScope(bool returnByRef, ref StorageClass stc)
+{
+    /* The old way was a determination based on whether the return
+     * value was by ref or not. The new way is `return scope` is set
+     * only if the `return` is immediately followed by `scope`, which
+     * is done by the parser.
+     * Keep the old way to not break pre-dip1000 code.
```

Comment:
How would this break pre-dip1000 code? Lifetime errors only appear with dip1000.

Reply:
Much of the scope code predates dip1000 and is enabled with or without the switch for it.

Reply:
That's new to me, can you give a code example of a scope error triggered without dip1000?

Reply:
see the attached test cases

Reply:
That makes no sense for two reasons:
- those tests have `REQUIRED_ARGS=-preview=dip1000`
- if they didn't, then your PR would break non `-preview=dip1000`, which is what you said you didn't do because of this